mlflow/models/evaluation/default_evaluator.py

```diff
@@ -475,8 +480,22 @@ def _evaluate_custom_metric(custom_metric_tuple, eval_df, builtin_metrics):
     if metric is None:
         raise MlflowException(f"{exception_header} returned None.")
 
-    if not _is_numeric(metric):
-        raise MlflowException(f"{exception_header} did not return a scalar numeric value.")
+    if not isinstance(metric, MetricValue):
```

Comment:
Should we add something like
`if _is_numeric(metric):
        return MetricValue(aggregate_results={custom_metric_tuple.name: metric})`
so that old custom metric fns will still work here? (if they don't access builtin metrics)